web3/src/commonMain/kotlin/dev.icerock.moko.web3/Web3.kt

```diff
@@ -20,144 +21,97 @@ import io.ktor.http.content.OutgoingContent
 import io.ktor.http.content.TextContent
 import kotlinx.serialization.KSerializer
 import kotlinx.serialization.builtins.serializer
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.encodeToString
 import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonArray
 import kotlinx.serialization.json.JsonElement
-import kotlinx.serialization.json.JsonPrimitive
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.encodeToJsonElement
+import kotlinx.serialization.json.int
+import kotlinx.serialization.json.jsonObject
+import kotlinx.serialization.json.jsonPrimitive
 
 class Web3(
     private val httpClient: HttpClient,
     private val json: Json,
     private val infuraUrl: String
 ) {
-    suspend fun getTransaction(transactionHash: TransactionHash): Transaction {
-        val requestSerializer = InfuraRequest.serializer(String.serializer())
-        val request: InfuraRequest<String> = InfuraRequest(
-            method = "eth_getTransactionByHash",
-            params = listOf(transactionHash.value)
-        )
-
-        val response: HttpResponse = httpClient.post {
-            url(infuraUrl)
-            body = json.encodeToJsonElement(requestSerializer, request).outgoingContent
-        }
-
-        return processResponse(
-            request = request,
-            serializer = InfuraResponse.serializer(Transaction.serializer()),
-            content = response.readText()
-        )
-    }
-
-    suspend fun getTransactionReceipt(transactionHash: TransactionHash): TransactionReceipt {
-        val requestSerializer = InfuraRequest.serializer(String.serializer())
-        val request: InfuraRequest<String> = InfuraRequest(
-            method = "eth_getTransactionReceipt",
-            params = listOf(transactionHash.value)
-        )
-
-        val response: HttpResponse = httpClient.post {
-            url(infuraUrl)
-            body = json.encodeToJsonElement(requestSerializer, request).outgoingContent
-        }
+    suspend fun getTransaction(
+        transactionHash: TransactionHash
+    ): Transaction = executeBatch(Web3Requests.getTransaction(transactionHash))[0]
```

Comment:
what if a list would be empty? I suggest to use `first/firstOrNull` here.

Reply:
Since this is an RPC request, server guarantee to return exact amount of responses as much as was requested. 

executeBatch(request(), request()) will always return a list of 2 elements if http success.
executeBatch(request(), request(), request()) will always return a list of 3 elements, etc.